pymc/step_methods/compound.py

```diff
@@ -126,6 +191,20 @@ def stop_tuning(self):
             self.tune = False
 
 
+def get_stats_dtypes_shapes_from_steps(
+    steps: Iterable[BlockedStep],
+) -> Dict[str, Tuple[StatDtype, StatShape]]:
+    """Combines stats dtype shape dictionaries from multiple step methods.
+
+    In the resulting stats dict, each sampler stat is prefixed by `sampler_#__`.
+    """
+    result = {}
+    for s, step in enumerate(steps):
+        for sname, (dtype, shape) in step.stats_dtypes_shapes.items():
+            result[f"sampler_{s}__{sname}"] = (dtype, shape)
```

Comment:
Currently these dictionary keys are not passed to arviz when creating InferenceData objects. But if/when they are, we'll probably want a way for the user to map back from `<step name>` to `<list of variables that were sampled by that stepper>`

Reply:
Agreed. ArviZ should add such a field, but even before that gets done we can add this to `mcbackend.RunMeta`